pkg/apis/keepalive/v1beta1/register.go

```diff
@@ -0,0 +1,74 @@
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// VirtualIP is a keepalived CRD specificiation for virtual IP addresses
+type VirtualIP struct {
```

Comment:
Is the intent with this CRD that every IP + interface -> Service:port mapping would have a separate definition? That would require quite a bit duplication by the user if they had multiple services on one IP. What about a way to select which services apply by labels?

Reply:
> Is the intent with this CRD that every IP + interface -> Service:port mapping would have a separate definition?

I changed the definition to a list of serviceReferences

Reply:
What about instead of binding the virtual IP to services in a centralized spot in the CRD, we decentralize it to be an annotation on the service? The annotation would be reference back the CRD's `metadata.name`. For example:

```yaml
apiVersion: v1
kind: Service
metadata:
  name: myservice
  annotations:
    keepalived/vip: myvipname
spec:
  externalIPs:
  - 10.0.22.170
```

Reply:
It'd be even nicer if it could populate the externalIPs on the service object. MetalLB does something similar to what I'm suggesting using annotations and modifying service IP:

https://metallb.universe.tf/usage/

Reply:
I am aware of the `externalIPs` field but using that also makes validation harder (I am thinking to add a webhook to avoid invalid configuration/duplicated IPs). Also using the service could lead to someone ask for `externalTrafficPolicy` support

Reply:
I think people should be aware that `externalTrafficPolicy` does not apply to externalIPs. We can document it in the README if need be or point to kubernetes docs.

If populating externalIPs is difficult/error prone, what do you think about just the annotation idea to do service binding?